scripts/ci/compiler-hash.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+COMPILER=$1
+
+$COMPILER -march=native -E -v - < /dev/null 2>&1 | grep cc1 \
```

Comment:
The force-recache version is duplicated in the matrix. I think I will leave it there for a while for a period until I give the run on fail yaml programming. Is this ok?

Reply:
We can leave it as is in the interim, but ultimately, the desired workflow is
1. Try a build with ccache, using the cache with key X
2. If job-1 fails, run with force-recache, which should repair the cache with key X. If job-1 succeeds, skip the recache job.
This would happen for the minimal/full compile variants.